rasa/utils/tensorflow/models.py

```diff
@@ -263,14 +264,20 @@ def rasa_predict(self, model_data: RasaModelData) -> Dict[Text, tf.Tensor]:
         batch_in = RasaBatchDataGenerator.prepare_batch(model_data.data)
 
         if self._run_eagerly:
-            return tf_utils.to_numpy_or_python_type(self.predict_step(batch_in))
+            outputs = tf_utils.to_numpy_or_python_type(self.predict_step(batch_in))
+            if np.size(outputs[DIAGNOSTIC_DATA]["attention_weights"]) == 0:
+                outputs[DIAGNOSTIC_DATA]["attention_weights"] = None
+            return outputs
 
         if self._tf_predict_step is None:
             self._tf_predict_step = tf.function(
                 self.predict_step, input_signature=self._dynamic_signature(batch_in)
             )
 
-        return tf_utils.to_numpy_or_python_type(self._tf_predict_step(batch_in))
+        outputs = tf_utils.to_numpy_or_python_type(self._tf_predict_step(batch_in))
+        if np.size(outputs[DIAGNOSTIC_DATA]["attention_weights"]) == 0:
```

Comment:
could you make it general, and return `None` instead of any empty array? regardless of the key

Reply:
This is the thing; I didn't want to make it general because I don't know what different things we do with the outputs. But I can try and see if something actually breaks...

Reply:
hm. Just do it within `DIAGNOSTIC_DATA` then

Reply:
Okay, this should now be fixed.